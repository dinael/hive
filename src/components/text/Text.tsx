import { FC } from 'react'
import styles from './Text.module.scss'

export type TextProps = {
  children: any
  block?: boolean
  ellipsis?: boolean
  a11yLength?: boolean
  className?: string
}

export const Text: FC<TextProps> = ({
  children ='',
  block = false,
  ellipsis = false,
  a11yLength = false,
  className,
  ...props
}: TextProps) => {

  const textStyles = `
    ${styles['text-component']}
    ${ellipsis ? styles['ellipsis'] : ''}
    ${className || ''}
  `

  const Tag = `${!block ? 'span' : 'p' }` as keyof JSX.IntrinsicElements

  return (
    <Tag className={textStyles} {...props}>
        { children }
    </Tag>
  )
}

export default Text
