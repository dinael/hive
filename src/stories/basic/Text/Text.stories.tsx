import { Text, TextProps } from '@/components/text/Text'

export default {
  title: 'Components/Basic/Text',
  component: Text,
  tags: ['autodocs'],
  parameters: {
    vitest: {
      testFile: "Text.test.tsx"
    }
  },
  argTypes: {
    children: { name: 'Text', control: 'text' },
    block: { name: 'Block', control: 'boolean' },
    ellipsis: {name: 'Ellipsis', control: 'boolean' },
    className: {name: 'className', control: 'text'},
  }
}

export const Default = (args: TextProps) => ([
  <Text {...args} />
])

export const withChildren = (args: TextProps) => ([
  <Text {...args}>
    This text is a children element
  </Text>
])

export const Block = (args: TextProps) => ([
  <Text {...args} />
])

Default.args = {
  text: 'Hello World',
  block: false,
  ellipsis: false,
  a11yLength: false,
  className: '',
}

withChildren.args = {
  ellipsis: false,
  a11yLength: false,
  className: '',
  block: true,
}

Block.args = {
  ...Default.args,
  text: 'Hello World, this a text block is a tag p',
  block: true,
}

