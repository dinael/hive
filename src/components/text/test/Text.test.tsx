import React from 'react'
import { describe, test, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import { Text, TextProps } from '../Text'

describe('Text', () => {
  const defaultProps: TextProps = {
    children: 'Example text',
    block: false,
    className: 'className-test',
  }

  test('renders snapshot', () => {
    const { container } = render(<Text {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  test('renders with the correct tag name', () => {
    const { container } = render(<Text {...defaultProps} />)
    const textElement = container.querySelector('span')
    expect(textElement?.tagName).toBe('SPAN')
    expect(container).toMatchSnapshot()
  });

  test('renders with the correct text', () => {
    const { container } = render(<Text {...defaultProps} />)
    const textElement = container.querySelector('span')
    expect(textElement?.textContent).toBe('Example text')
    expect(screen.getByText('Example text'))
    expect(container).toMatchSnapshot()
  });

  test('renders with the correct extra className', () => {
    const { container } = render(<Text {...defaultProps} />)
    const textElement = container.querySelector('span')
    expect(textElement?.className).toContain('className-test')
    expect(container).toMatchSnapshot()
  });

  test('renders as a block element', () => {
    const { container } = render(<Text {...defaultProps} block={true} />)
    const textElement = container.querySelector('p')
    expect(textElement).not.toBeNull()
    expect(textElement?.textContent).toBe('Example text')
    expect(screen.getByText('Example text'))
    expect(container).toMatchSnapshot()
  });

})
