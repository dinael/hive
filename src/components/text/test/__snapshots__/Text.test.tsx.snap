// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > Text renders as a block element 1`] = `
<div>
  <p
    class="
    text-component
    
    className-test
  "
  >
    Example text
  </p>
</div>
`;

exports[`Text > Text renders snapshot 1`] = `
<div>
  <span
    class="
    text-component
    
    className-test
  "
  >
    Example text
  </span>
</div>
`;

exports[`Text > Text renders with the correct extra className 1`] = `
<div>
  <span
    class="
    text-component
    
    className-test
  "
  >
    Example text
  </span>
</div>
`;

exports[`Text > Text renders with the correct tag name 1`] = `
<div>
  <span
    class="
    text-component
    
    className-test
  "
  >
    Example text
  </span>
</div>
`;

exports[`Text > Text renders with the correct text 1`] = `
<div>
  <span
    class="
    text-component
    
    className-test
  "
  >
    Example text
  </span>
</div>
`;
